### PR TITLE
Correct serialized property name on AsciiFoldingTokenFilter

### LINKED ISSUE #2892: [asciifolding] "preserve_original" JsonProperty
"preserve_original": true ends up being "preserveOriginal": true, thus breaking the functionality.

This is still an issue, it was referenced #2786 here, a solution was suggested but no merge was made. Could you take a look on it please?

### PR DESCRIPTION
**Pull request by russcam**
Closes #2892

### CODE REVIEW

File: src/Tests/CodeStandards/Analysis.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using FluentAssertions;
+using Nest;
+using Newtonsoft.Json;
+using Tests.Framework;
+
+namespace Tests.CodeStandards
```
Comment (Mpdreamz) [inline src/Tests/CodeStandards/Analysis.cs]: 👍 💯

File: src/Nest/Analysis/Analyzers/AnalyzerBase.cs [hunk 2]
```diff
@@ -5,21 +5,18 @@ namespace Nest
...
 
 	public abstract class AnalyzerBase : IAnalyzer
 	{
 		internal AnalyzerBase() { }
 
-		protected AnalyzerBase(string type)
-		{
-			Type = type;
-		}
+		protected AnalyzerBase(string type) => Type = type;
```
Comment (Mpdreamz) [inline src/Nest/Analysis/Analyzers/AnalyzerBase.cs]: I better upgrade Rider to 2017.2 and visual studio latest preview :) roll on c# 7.1

**Comment by codebrain** [comment]
Backported to 2.x and labelled 2.5.7

**Comment by codebrain** [comment]
Backported to 5.x and labelled 5.6.0

**Pull request merged**